### PR TITLE
[fix #100] support current pd's safepoint update interface

### LINKED ISSUE #100: [gc-worker] support current pd safepoint update interface
## Feature Request

**Is your feature request related to a problem? Please describe:**

support current pd safepoint update interface

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by haojinming**
### What problem does this PR solve?

Issue Number: close #100 

Problem Description:

### What is changed and how does it work?

support current pd's safepoint update interface
UpdateServiceGCSafePoint
UpdateGCSafePoint

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Manual test (add detailed scripts or steps below)

### Side effects

- No side effects

### Related changes

- No related changes

### CODE REVIEW

File: gc-worker/server/server_test.go [hunk 1]
```diff
@@ -108,3 +158,62 @@ func NewTestSingleConfig() *embed.Config {
...
+
+	newSp = s.calcNewGCSafePoint(1000, 100)
+	require.Equal(t, newSp, uint64(100))
+
+	newSp = s.calcNewGCSafePoint(1000, 10000)
+	require.Equal(t, newSp, uint64(1000))
+
+}
+
+func TestCalcGcSafePoint(t *testing.T) {
```
Comment (pingyu) [inline gc-worker/server/server_test.go:182]: Suggest to make naming be consistent.

```suggestion
func TestCalcGCSafePoint(t *testing.T) {
```

File: gc-worker/server/server_test.go [hunk 2]
```diff
@@ -108,3 +158,62 @@ func NewTestSingleConfig() *embed.Config {
...
+	curTs := tsoutil.GenerateTS(tsoutil.GenerateTimestamp(time.Now(), 0))
+	expectTs := time.Now().Add(-cfg.GCLifeTime.Duration)
+	expectGcSafePoint := tsoutil.GenerateTS(tsoutil.GenerateTimestamp(expectTs, 0))
+	gcSafePoint, err := s.getGCWorkerSafePoint(ctx)
+	require.NoError(t, err)
+	require.LessOrEqual(t, expectGcSafePoint, gcSafePoint)
+	require.LessOrEqual(t, gcSafePoint, curTs)
+}
+
+func TestUpdateGcSafePoint(t *testing.T) {
```
Comment (pingyu) [inline gc-worker/server/server_test.go:203]:
```suggestion
func TestUpdateGGSafePoint(t *testing.T) {
```

**Review by pingyu — COMMENTED** [review]
Rest LGTM~

File: gc-worker/server/server_test.go (hunk 1 above)
Comment (haojinming) [reply gc-worker/server/server_test.go]: done

File: gc-worker/server/server_test.go (hunk 2 above)
Comment (haojinming) [reply gc-worker/server/server_test.go]: done

**Review by pingyu — APPROVED** [review]
LGTM~

**Review by ystaticy — APPROVED** [review]

**Comment by ystaticy** [comment]
LGTM~

**Review by ystaticy — APPROVED** [review]

**Pull request merged**